### ISSUE TITLE
Oni can't display the contents of any file(including tutorials)

### ISSUE THREAD

**Issue by rektrex** [onivim/oni#2704]
**Oni Version:** 0.3.9
**Neovim Version (Linux only):** 0.3.2
**Operating System:** Arch Linux(Linux 4.20.0)

**Issue:** Oni can't display the contents of any file(including tutorials) or a new file. Tested with and without an existing init.vim, unmodified oni config, and no plugins.Other UI elements seem to work fine. 

**Screenshot after opening "Main.hs"**
![Screenshot after opening "Main.hs"](https://user-images.githubusercontent.com/23721797/50599382-f30b4080-0ed3-11e9-958a-56823bcaff28.png)

**Expected behavior:** The contents of the file is displayed.

**Actual behavior:** The editor pane is empty.

**Steps to reproduce:** 

1. Open Oni.
2. Open a new file or an existing file or a tutorial lesson.

**Comment by mbitokhov** [comment]
I'm new to the project, but I thought I'd take a look to see why. I cloned, built, and ran this project in debug mode to see if it'll give any better information. The only thing I found was an unhandled promise rejection in "TutorialManager"

```
TutorialManager.ts:135 Uncaught (in promise) TypeError: Cannot read property 'getLayerById' of undefined
    at TutorialManager.<anonymous> (TutorialManager.ts:135)
    at Generator.next (<anonymous>)
    at fulfilled (TutorialGameplayManager.ts:120)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188)
```

I can only assume there is a promise rejection elsewhere causing it to not open files, but this is the only place where it's unhandled. I spent a little bit of time looking around the project to try to better understand it

After following all of the function calls, I tracked it down to [this line](https://github.com/onivim/oni/blob/master/browser/src/Editor/NeovimEditor/NeovimEditor.tsx#L953), which [calls the getter `activeBuffer`](https://github.com/onivim/oni/blob/master/browser/src/Editor/NeovimEditor/NeovimEditor.tsx#L162), which [gets the `lastBufferId`](https://github.com/onivim/oni/blob/master/browser/src/Editor/NeovimEditor/NeovimEditor.tsx#L132) to retrieve the lastBufferId from the bufferManager. Only issue is, after a quick console.log, lastBufferId is null

It's getting a little bit late for me, I'll see if I can find out why this weekend.

**Comment by chuckmasucci** [comment]
Same for me running Manjaro and i3 on 0.3.9

**Comment by Honason** [comment]
I also can't see content, but only in tutorials. MacOS Mojave.

**Comment by gutuva** [comment]
Same for me: `Linux 4.20.0-arch1-1-ARCH` | `oni 0.3.9-1` | `neovim 0.3.3-1` | :(

**Comment by ray-x** [comment]
I'm getting the same blank editor behaviour in oni 0.3.7 (also tried oni 0.3.9)  and neovim 0.3.10 on ubuntu 18.10.

**Comment by delkano** [comment]
Following @mbitokhov findings, I've opened a bug in Neovim: https://github.com/neovim/neovim/issues/9473

**Comment by justinmk** [comment]
https://github.com/onivim/oni/pull/2613 is still pending. As explained there, Oni should call `nvim_ui_attach` after starting `nvim --embed`.

`nvim_get_mode` is (currently) "best effort", but inherently racy: if `nvim_get_mode` request is sent at just the "right" time then it could block. The best solution is to remove the need for `nvim_get_mode`  , which is what https://github.com/onivim/oni/pull/2613 should do.

**Comment by mbitokhov** [comment]
@justinmk 

Thanks for your input! I don't know why that PR wasn't pulled in yet. But I'm glad there's some reasonable explanation

**Comment by Akin909** [comment]
@mbitokhov @rektrex, @CrossR is currently looking into this (*I think*) and reached a similar conclusion, sorry for the delay re. a response and a resolution, we're currently spending a lot of time working on `Oni2` and the bits we need to get setup for that but this issue is a pretty big one we'd like to sort out

**Comment by mbitokhov** [comment]
@CrossR 

Let me know if there's some way I can help. I don't know much about the nvim API. But I might be able to get something done on a weekend.

**Comment by devendor** [comment]
This happens with several other GUIs on ubuntu bionic, nvim >=0.3.2. 

In troubleshooting the same behavior in eovim, it looks like a blocking/empty vimenter request was changed or added in 0.3.2 to eliminate spurious UI events during initialization. I believe you are expected to send a command back after your initialization is complete, otherwise it's just a notice that UI events are being blocked waiting on your initialization to complete.

Below, is a representation of the offending message that eovim chokes on after running with an option to dump msgpack content.

```
--------
[0, 1, "vimenter", []]
--------
ERR<26988>:eovim /home/me/share/util/src/eovim/src/nvim.c:360 _nvim_received_data_cb() Invalid message identifier 0
```

I have full traces of working vs non-working neovim versions and more explanation over here:

 https://github.com/jeanguyomarch/eovim/issues/38

**Comment by CrossR** [comment]
I think I've got this working now, in that it will launch normally and will launch when there is any startup messages as well.

I'm still trying to work around setting the colour scheme up correctly (since I've had to move the first draw call around, the colour scheme isn't set so there is a split second flash of black before the theme kicks in).

@devendor , there is an explanation of this change over in NeoVims wiki : https://github.com/neovim/neovim/wiki/Following-HEAD#20180922 which might make things a bit clearer on why it was changed, why its a good thing and how to work around it.
